tests/backward_compatibility/test_compatibility.py

```diff
@@ -43,7 +50,6 @@ def test_compatibility(tag, change_test_dir):
     Test that we still can analyze wanted versions without error
     """
 
-    os.system("rm -rf vehicle_signal_specification")
     os.system("git clone --depth 1 --branch " + tag +
```

Comment:
you could move this to the fixture above the yield so it is only cloned once in the beginning, no?

Reply:
but I just realized that that would mean you would have to change the scope of the `delete_files` fixture function to `"module"`..

Reply:
or would it be `"package"`, I'm currently not sure https://docs.pytest.org/en/6.2.x/fixture.html

Reply:
and obviously switch branch depending on tag, but please ignore if this creates any overhead

Reply:
Would indeed be doable. Lets see if I get time. As we only have one test case here the advantage might be limited, but we could possibly use it as playground and later reuse it for other files

Reply:
I did some tests, the `monkeypatch.chdir` command has function scope, but even if refactoring like below it is faster to do 3 shallow clones for a single branch than doing a full clone. So it does not seem to be worth the effort right now

```
@pytest.fixture(scope="module", autouse=True)
def initiate_test_env(request):
    os.chdir(request.fspath.dirname)
    os.system("git clone https://github.com/COVESA/vehicle_signal_specification")
    yield None
    os.system("rm -rf out.json out.txt vehicle_signal_specification")
    os.chdir(request.config.invocation_params.dir)
```